packages/cli/src/lib/autorun/autorun-command.unit.test.ts

```diff
@@ -0,0 +1,117 @@
+import { bundleRequire } from 'bundle-require';
+import { vol } from 'memfs';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import {
+  PortalUploadArgs,
+  ReportFragment,
+  uploadToPortal,
+} from '@code-pushup/portal-client';
+import { collectAndPersistReports } from '@code-pushup/core';
+import { report } from '@code-pushup/models/testing';
+import { DEFAULT_CLI_CONFIGURATION } from '../../../test/constants';
+import { yargsCli } from '../yargs-cli';
+import { yargsAutorunCommandObject } from './autorun-command';
+
+// This in needed to mock the API client used inside the upload function
+vi.mock('@code-pushup/portal-client', async () => {
+  const module: typeof import('@code-pushup/portal-client') =
+    await vi.importActual('@code-pushup/portal-client');
+
+  return {
+    ...module,
+    uploadToPortal: vi.fn(
+      async () => ({ packageName: '@code-pushup/cli' } as ReportFragment),
+    ),
+  };
+});
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+vi.mock('@code-pushup/core', async () => {
+  const core = await vi.importActual('@code-pushup/core');
+  return {
+    ...(core as object),
+    collectAndPersistReports: vi.fn().mockResolvedValue({}),
+  };
+});
+
+// Mock bundleRequire inside importEsmModule used for fetching config
+vi.mock('bundle-require', async () => {
+  const { minimalConfig }: typeof import('@code-pushup/models/testing') =
+    await vi.importActual('@code-pushup/models/testing');
+  return {
+    bundleRequire: vi
+      .fn()
+      .mockResolvedValue({ mod: { default: minimalConfig() } }),
+  };
+});
+
+describe('autorun-command', () => {
+  beforeEach(async () => {
```

Comment:
There's no `await`, so no need for `async`. (We have a lint rule for this, but errors need fixing first.)

```suggestion
  beforeEach(() => {
```

Reply:
Nice catch!